i2c-testsuite/tests/test.rs

```diff
@@ -0,0 +1,46 @@
+#![no_std]
+#![no_main]
+
+use defmt::unwrap;
+use defmt_rtt as _; // global logger
+use embedded_time::rate::Hertz;
+use panic_probe as _;
+use stm32wl_hal::{
+    embedded_hal::blocking::i2c::WriteRead,
+    gpio::{pins, PortB},
+    i2c::I2c1,
+    pac, rcc,
+};
+
+#[defmt_test::tests]
+mod tests {
+    use super::*;
+
+    #[init]
+    fn init() -> I2c1<(pins::B8, pins::B7)> {
+        let mut dp: pac::Peripherals = unwrap!(pac::Peripherals::take());
+
+        rcc::set_sysclk_to_msi_48megahertz(&mut dp.FLASH, &mut dp.PWR, &mut dp.RCC);
+        let gpiob = PortB::split(dp.GPIOB, &mut dp.RCC);
+        I2c1::new(
+            dp.I2C1,
+            (gpiob.pb8, gpiob.pb7),
+            Hertz(100_000),
+            &mut dp.RCC,
+            false,
+        )
+    }
+
+    #[test]
+    fn sht31_measurement(i2c: &mut I2c1<(pins::B8, pins::B7)>) {
+        defmt::warn!("A SHT31 sensor must be connected to the board on pins PB8 (SCL) & PB7 (SDA) for this test to work");
```

Comment:
Normally I would buy this sensor, but it is very difficult to get parts quickly these days.

One idea I had was to use two I2C bus' shorted together and configure one as the secondary, no additional hardware requirements then.
I tried it out, and it does work pretty well: https://github.com/newAM/stm32wl-hal/blob/i2c-loopback-test/i2c-testsuite/tests/test.rs#L91

Feel free to include that as part of this PR, or I can add it after, I do want something that can run automatically for on-target CI.

Reply:
OK, I've put the loopback test there too. I'll test later today with the board too. Do you want me to remove the sht31 test, or can it be skipped by the test suite (I should probably learn how that works 😅)?

Reply:
The SHT31 test is fine, it does not error out without it connected so that's fine by me.